### ISSUE TITLE
Azure App Service: Could not load file or assembly Microsoft.Data.SqlClient.SNI.x64.dll, The module was expected to contain an assembly manifest.

### ISSUE THREAD

**Issue by cvb941** [dotnet/SqlClient#669]
### Describe the bug
Our project is working fine on the developer machine. However, when deployed to Azure App Service, the throws the following exception during initializing.

```
Exception message: Could not load file or assembly 'file:///D:\home\site\wwwroot\bin\Microsoft.Data.SqlClient.SNI.x64.dll' or one of its dependencies. The module was expected to contain an assembly manifest.
Stack trace:
[BadImageFormatException: Could not load file or assembly 'file:///D:\home\site\wwwroot\bin\Microsoft.Data.SqlClient.SNI.x64.dll' or one of its dependencies. The module was expected to contain an assembly manifest.]
   System.Reflection.RuntimeAssembly._nLoad(AssemblyName fileName, String codeBase, Evidence assemblySecurity, RuntimeAssembly locationHint, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean throwOnFileNotFound, Boolean forIntrospection, Boolean suppressSecurityChecks) +0
   System.Reflection.RuntimeAssembly.nLoad(AssemblyName fileName, String codeBase, Evidence assemblySecurity, RuntimeAssembly locationHint, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean throwOnFileNotFound, Boolean forIntrospection, Boolean suppressSecurityChecks) +36
   System.Reflection.RuntimeAssembly.InternalLoadAssemblyName(AssemblyName assemblyRef, Evidence assemblySecurity, RuntimeAssembly reqAssembly, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean throwOnFileNotFound, Boolean forIntrospection, Boolean suppressSecurityChecks) +152
   System.Reflection.RuntimeAssembly.InternalLoadFrom(String assemblyFile, Evidence securityEvidence, Byte[] hashValue, AssemblyHashAlgorithm hashAlgorithm, Boolean forIntrospection, Boolean suppressSecurityChecks, StackCrawlMark& stackMark) +102
   System.Reflection.Assembly.LoadFrom(String assemblyFile) +34
   WebActivator.PreApplicationStartCode.Start() in D:\Code\Bitbucket\WebActivator\WebActivator\PreApplicationStartCode.cs:11
```

### Further technical details
Microsoft.Data.SqlClient version: 2.0.0
.NET target: Framework 4.7.2

**Additional context**
Microsoft.Data.SqlClient.SNI version: 2.0.0

**Comment by Wraith2** [comment]
That dll isn't a managed assembly it's native code. As such whatever is trying to load it as an assembly shouldn't be.

**Comment by cheenamalhotra** [comment]
Hi @cvb941 

Do you have a repro or could you explain more about your deployment process?

**Comment by cvb941** [comment]
Hello everyone and thanks for the replies.

I investigated a bit more and while trying to isolate the issue, I created a new App Service where, surprisingly, the issue wasn't present.

I do not know the insides of how App Service deployment works but I figured that it doesn't really clean the instance on new deployment and there were some remnants that caused the issue with the new release.

I fixed the issue on the original App Service instance by discovering and specifying the option 'Remove additional files at destination' during deployment, which is the "Clean" functionality I was looking for. Maybe it will help someone else with a similar issue.

As for what specifically caused the issue in the first place, I guess will remain a mystery :)
